examples/mac/dump_tree_axapi.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+import argparse
+import sys
+import acacia_axapi
+
+ATTRIBUTES = [
+  'AXRoleDescription',
+  'AXTitle',
+  'AXDescription',
+  'AXFocused',
+]
+
+
+def print_node_and_subtree(node, level):
+  prefix = "--" * level
+  role_and_attributes = serialize_role_and_attributes(node, ATTRIBUTES)
+  print(f'{prefix} {role_and_attributes}')
+
+  if not node.hasAttribute('AXChildren'):
+    return
+
+  children = get_list_attribute_value(node, 'AXChildren')
+  for child in children:
+    print_node_and_subtree(child, level + 1)
+
+
+def serialize_role_and_attributes(node, attributes):
+  output = get_attribute_value(node, 'AXRole')
+
+  for attribute in attributes:
+    if not node.hasAttribute(attribute):
+      continue
+
+    value = get_attribute_value(node, attribute)
+    match type(value).__name__:
+      case 'NoneType':
+        continue
+      case 'bool':
+        if value:
+          output += f' {attribute}'
+      case 'str':
+        value_utf16 = value.encode('utf-16', 'surrogatepass').decode('utf-16', 'replace')
+        output += f' {attribute}="{value_utf16}"'
+      case _:
+        output += f' {attribute}={value}'
+
+  return output
+
+
+def get_attribute_value(node, attribute):
+  try:
+    type = node.getValueType(attribute)
+    match type:
+      case acacia_axapi.ValueType_LIST:
+        return get_list_attribute_value(node, attribute)
+      case acacia_axapi.ValueType_BOOLEAN:
+        return node.getBooleanValue(attribute)
+      case acacia_axapi.ValueType_INT:
+        return node.getIntValue(attribute)
+      case acacia_axapi.ValueType_FLOAT:
+        return node.getFloatValue(attribute)
+      case acacia_axapi.ValueType_STRING:
+        return node.getStringValue(attribute)
+      case acacia_axapi.ValueType_URL:
+        return node.getURLValue(attribute)
+      case acacia_axapi.ValueType_NODE:
+        return node.getNodeValue(attribute)
+      case acacia_axapi.ValueType_POINT:
+        return node.getPointValue(attribute)
+      case acacia_axapi.ValueType_SIZE:
+        return node.getSizeValue(attribute)
+      case acacia_axapi.ValueType_RECT:
+        return node.getRectValue(attribute)
+      case acacia_axapi.ValueType_RANGE:
+        return node.getRangeValue(attribute)
+      case acacia_axapi.ValueType_DICTIONARY:
+        return node.getDictionaryValue(attribute)
+      case _:
+        type_string = acacia_axapi.ValueTypeToString(type)
+        print(f'Unsupported type: {type_string} for {attribute}')
+        return None
+  except Exception as error:
+    return f'(error: {error})'
+
+
+def get_list_attribute_value(node, attribute):
+  count = node.getListElementCount(attribute)
+  if count == 0:
+    return []
+
+  type = node.getListElementType(attribute)
+  list = None
+  match type:
+    case acacia_axapi.ValueType_NODE:
+      list = node.getNodeListValue(attribute)
+    case acacia_axapi.ValueType_STRING:
+      list = node.getStringListValue(attribute)
```

Comment:
It took me a bit to figure out, but eventually I realised it's (for some reason) because those attribute getters took a `std::string&` rather than a `const std::string&` and somehow that caused an error when converting the Python attribute name string to the C++ string. Seems to work now! Thanks so much for catching this!